core/src/test/java/org/elasticsearch/search/suggest/completion/CategoryContextMappingTests.java

```diff
@@ -116,6 +118,105 @@ public void testIndexingWithSimpleContexts() throws Exception {
+        assertContextSuggestFields(fields, 3);
+    }
+    
+    public void testIndexingWithSimpleBooleanContexts() throws Exception {
+        String mapping = jsonBuilder().startObject().startObject("type1")
+                .startObject("properties").startObject("completion")
+                .field("type", "completion")
+                .startArray("contexts")
+                .startObject()
+                .field("name", "ctx")
+                .field("type", "category")
+                .endObject()
+                .endArray()
+                .endObject().endObject()
+                .endObject().endObject().string();
+
+        DocumentMapper defaultMapper = createIndex("test").mapperService().documentMapperParser().parse("type1", new CompressedXContent(mapping));
+        FieldMapper fieldMapper = defaultMapper.mappers().getMapper("completion");
+        MappedFieldType completionFieldType = fieldMapper.fieldType();
+        ParsedDocument parsedDocument = defaultMapper.parse("test", "type1", "1", jsonBuilder()
+                .startObject()
+                .startArray("completion")
+                .startObject()
+                .array("input", "suggestion5", "suggestion6", "suggestion7")
+                .startObject("contexts")
+                .field("ctx", true)
+                .endObject()
+                .field("weight", 5)
+                .endObject()
+                .endArray()
+                .endObject()
+                .bytes());
+        IndexableField[] fields = parsedDocument.rootDoc().getFields(completionFieldType.name());
+        assertContextSuggestFields(fields, 3);
+    }
+    
+    public void testIndexingWithSimpleNULLContexts() throws Exception {
+        String mapping = jsonBuilder().startObject().startObject("type1")
+                .startObject("properties").startObject("completion")
+                .field("type", "completion")
+                .startArray("contexts")
+                .startObject()
+                .field("name", "ctx")
+                .field("type", "category")
+                .endObject()
+                .endArray()
+                .endObject().endObject()
+                .endObject().endObject().string();
+
+        DocumentMapper defaultMapper = createIndex("test").mapperService().documentMapperParser().parse("type1", new CompressedXContent(mapping));
+        try {
+            defaultMapper.parse("test", "type1", "1", jsonBuilder()
+                .startObject()
+                .startArray("completion")
+                .startObject()
+                .array("input", "suggestion5", "suggestion6", "suggestion7")
+                .startObject("contexts")
+                .nullField("ctx")
+                .endObject()
+                .field("weight", 5)
+                .endObject()
+                .endArray()
+                .endObject()
+                .bytes());
+        } catch(MapperParsingException e) {
```

Comment:
You can use `expectThrows(MapperParsingException.class, () -> { /* code that throws */});`, then check the message on the returned exception.